request.session type is always any

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Fastify version

4.10.2

### Plugin version

10.1.0

### Node.js version

18.12.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.1

### Description

Even after adding a type declaration like
```typescript
declare module "fastify" {
    interface Session {
        user: User
    }
}
```
to the code the type of request.session is still `any`.

<img width="407" alt="Screenshot 2022-12-22 at 10 56 22 AM" src="https://user-images.githubusercontent.com/691152/209206847-8a73b9e5-aeb7-4210-a175-1d3bb640e5e3.png">


### Steps to Reproduce

start an empty fastify project, add fastify-session and follow https://github.com/fastify/session#typescript-support

### Expected Behavior

the session type should be applied to request.session

@kylerush thanks to you example I was able to track this down to the `esModuleInterop` typescript setting. If you set it to `false` in your repo the issue starts to reproduce. If you also make sure tsc checks libs (`"skipLibCheck": false`) you can also see the error with tsc:

```
% ./node_modules/.bin/tsc
node_modules/@fastify/session/types/types.d.ts:14:22 - error TS2694: Namespace '"/Users/fabian/tmp/fastify-session-type-check/node_modules/fastify/fastify"' has no exported member 'Session'.

14     session: Fastify.Session;
                        ~~~~~~~

node_modules/@fastify/session/types/types.d.ts:29:60 - error TS2694: Namespace '"/Users/fabian/tmp/fastify-session-type-check/node_modules/fastify/fastify"' has no exported member 'Session'.

29 type CallbackSession = (err: Error | null, result: Fastify.Session) => void;
                                                              ~~~~~~~

node_modules/@fastify/session/types/types.d.ts:86:24 - error TS2694: Namespace '"/Users/fabian/tmp/fastify-session-type-check/node_modules/fastify/fastify"' has no exported member 'Session'.

86       session: Fastify.Session,
                          ~~~~~~~

node_modules/@fastify/session/types/types.d.ts:184:43 - error TS2694: Namespace '"/Users/fabian/tmp/fastify-session-type-check/node_modules/fastify/fastify"' has no exported member 'Session'.

184     constructor(map?: Map<string, Fastify.Session>);
                                              ~~~~~~~

node_modules/@fastify/session/types/types.d.ts:187:24 - error TS2694: Namespace '"/Users/fabian/tmp/fastify-session-type-check/node_modules/fastify/fastify"' has no exported member 'Session'.

187       session: Fastify.Session,
                           ~~~~~~~


Found 5 errors in the same file, starting at: node_modules/@fastify/session/types/types.d.ts:14
```
You should be able to reproduce the issue if you check out my branch https://github.com/kylerush/fastify-session-type-check/pull/1

--- a/types/types.d.ts
+++ b/types/types.d.ts
@@ -11,7 +11,7 @@ declare module 'fastify' {
 
   interface FastifyRequest {
     /** Allows to access or modify the session data. */
-    session: Fastify.Session;
+    session: Session;
 
     /** A session store. */
     sessionStore: Readonly<fastifySession.SessionStore>;

